---
title: "IMLS1"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{IMLS1}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
# Evolutionary distinctiveness
## IMLS update, 2021-06-23, Andrew Hipp
As we have been waiting for our final sequencing results, which just came in at the end of May [Hooray!], I have been thinking about how to organize and use the phylogenetic data. I'll present today `edivColl` -- for Evolutionary Diversity in Collections -- which is the start of an `R` package that will provide tools for us and for others doing similar work.

The package lives at https://github.com/andrew-hipp/edivColl, but it can be migrated to the shared repository if that's better, or we can fork it into there if we like... whatever is needed. At this point, I'm working in a separate repository b/c that's what I'm used to.

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
devtools::install_github('andrew-hipp/edivColl', dep = FALSE)
# note that if you don't have dependencies already installed, set dep = TRUE or
#   leave out the dep argument. it's in there now b/c updating dependencies is enforced
#   and sometimes irritating during development.
library(edivColl)
```
## First, let's get some phylogenies
Our phylogenies as they stand now are included as package data. My thought is that we will leave them as such unless there is some concern about doing so; I'd rather have the data all easily accessible to reviewers and users in the package instead of making them scrounge around in other repositories (which I and almost everyone else does; it's not the end of the world, I just don't think it's ideal if you can avoid it).

The phylogenies we are developing have lots of accession information appended, and multiple inviduals per species. So for example:

```{r}
library(ape) # this is a standard R phylogenetics package
data(malus_tr) # this is one of Lindsey's Malus trees
head(malus$tip.label, 10) # these are the tip labels as we build them
plot(malus, cex = 0.6)
```

For understanding phylogeny of the group, the accessions information and inclusion of multiple exemplars per species is essential. But for phylogenetic diversity and distinctiveness, this is most distracting. So let's strip off that information.

```{r}
malus2 <- cleanPhylo(malus, delim = '_|_')
head(malus2$tip.label, 20)
```

Isn't that nice?

Also, for phylogenetic diversity, we generally want the branch lengths to be in time units; this tree has its branches in units of substitutions / nucleotide. Let's make a rather casual assumption about the rate of evolution evolving along a tree and rescale the tree appropriately.

```{r}
malus2 <- chronos(malus2)
plot(malus2, cex = 0.6)
```
